agent/engine/dependencygraph/graph.go

```diff
@@ -109,6 +136,17 @@ func linksToContainerNames(links []string) []string {
 	return names
 }
 
+func executionCredentialsResolved(target *api.Container, id string, manager credentials.Manager) bool {
+	if target.GetKnownStatus() >= api.ContainerPulled ||
+		!target.ShouldPullWithExecutionRole() ||
+		target.GetDesiredStatus() == api.ContainerStopped {
```

Comment:
just to be safe, can you make this `target.GetDesiredStatus() >= api.ContainerStopped` ?